File: x_verba/_test_sample/sample.ts
import { generateText } from "ai";

interface Options {
  retries?: number;
  mode?: "fast" | "slow";
}

export async function runAgent(input: string, opts: Options = {}) {
  const isValid = validateInput(input);
  if (!isValid) {
    throw new Error("invalid input");
  } else if (opts.mode === "fast") {
    console.log("fast mode");
  }

  const label = isValid ? "ok" : "bad";

  switch (opts.mode) {
    case "fast":
      doFastThing();
      break;
    case "slow":
      doSlowThing();
      break;
    default:
      doDefaultThing();
  }

  for (const item of [1, 2, 3]) {
    console.log(item);
  }

  try {
    await fetch("https://api.example.com/data", { method: "POST" });
  } catch (err) {
    // swallow error silently
  }

  try {
    db.delete(input);
  } catch (e) {
    console.error("delete failed", e);
    notifyAdmin(e);
  }

  if (requireApproval(input)) {
    await agent.invoke(input);
  }

  const result = await generateText({ prompt: input });
  return result;
}

File: x_verba/_test_sample/Sample.cs
using OpenAI.Chat;

namespace AgentApp
{
    public class AgentRunner
    {
        public async Task<string> RunAgent(string input, string mode)
        {
            bool isValid = ValidateInput(input);
            if (!isValid)
            {
                throw new Exception("invalid input");
            }
            else if (mode == "fast")
            {
                Console.WriteLine("fast mode");
            }

            string label = isValid ? "ok" : "bad";

            switch (mode)
            {
                case "fast":
                    DoFastThing();
                    break;
                case "slow":
                    DoSlowThing();
                    break;
                default:
                    DoDefaultThing();
                    break;
            }

            foreach (var item in new[] { 1, 2, 3 })
            {
                Console.WriteLine(item);
            }

            try
            {
                await FetchDataAsync(input);
            }
            catch (Exception)
            {
            }

            try
            {
                Db.Delete(input);
            }
            catch (Exception e)
            {
                Logger.Error("delete failed", e);
                NotifyAdmin(e);
            }

            if (RequireApproval(input))
            {
                await Agent.InvokeAsync(input);
            }

            var client = new OpenAIClient("sk-test");
            var chatClient = client.GetChatClient("gpt-4");
            var result = await chatClient.CompleteChatAsync(input);

            return result.ToString();
        }
    }
}
